flink-libraries/flink-table/src/main/scala/org/apache/flink/table/plan/nodes/FlinkRel.scala

```diff
@@ -120,6 +120,7 @@ trait FlinkRel {
         case typeName if SqlTypeName.YEAR_INTERVAL_TYPES.contains(typeName) => s + 8
         case typeName if SqlTypeName.DAY_INTERVAL_TYPES.contains(typeName) => s + 4
         case SqlTypeName.TIME | SqlTypeName.TIMESTAMP | SqlTypeName.DATE => s + 12
+        case SqlTypeName.ARRAY | SqlTypeName.ROW | SqlTypeName.ANY => s + 24
```

Comment:
Any chance we can expand the ARRAY or the ROW type, and only counting the basic data structures to make this a little more accurate?